### DISCUSSION TITLE
Current status of Libplanet.Explorer.Executable and what should we do about it

### DISCUSSION THREAD

**Discussion by tkiapril** [planetarium/libplanet#2588] [General]
Currently, Libplanet.Explorer.Executable is designed in a way that it is agnostic about what the actions in the given chain look like, by providing a no-op action `NullAction` as `T` to the `BlockChain<T>` object. On action evaluation, this action did nothing and returned an empty state. This should have worked until the introduction of state root hashes, but now that the state root hash of each blocks are checked against the execution result, getting the correct state root hash requires Libplanet.Explorer.Executable to use the correct action types for each chain it hosts, and otherwise renders it unusable. We have to decide whether to decommission it, or make it work in other means. See also: #2243

**Comment by dahlia** [comment]
We probably need a standalone executable to run explorer in the long run, but IMHO the easiest way at present would be dropping *Libplanet.Explorer.Executable*.  Instead, we could properly implement light node (i.e., node with no action loaded) in the future.

**Comment by dahlia** [comment]
Even if we're going to drop *Libplanet.Explorer.Executable*, we need to move subcommand `Libplanet.Explorer.Executable schema` (see also: <https://github.com/planetarium/libplanet/pull/2563>) somewhere else.

Cc @moreal.
